### ISSUE TITLE
If you ask the service to send too many notifications in a quick succession it sends infinite notifications

### ISSUE THREAD

**Issue by mxcl** [PerfectlySoft/Perfect-Notifications#10]
Pretty sure this is not my code doing this.

If you ask the service to send like 50 notifications in quick succession it repeatedly sends the same batch of notifications forever.

I have seen this happen twice now.

**Comment by mxcl** [comment]
This is deployed to AWS on a fairly constrained server.

My send is within a router callback for `PerfectHTTPServer`, there is no loop in my code and I am logging when the callback happens, it only happens once. So unless Apple are doing this it does seem to be within PerfectNotifications or one of its dependencies.

**Comment by mxcl** [comment]
However notably the callback is called like 50 times in rapid succession. *separate instantiations* on different threads. Possibly a concurrency violation?

**Comment by kjessup** [comment]
Thank you for opening the issue. I just wanted to let you know that I'm digging into this one today and will report back ASAP.

**Comment by mxcl** [comment]
The recent commit fixes this, though an awful lot of notifications seem to get dropped instead, with a lot of internalServerError errors and `~HTTP2Client.startReadThread` messages.

So thanks, I'll look into whether or not I can help fix that.

**Comment by mxcl** [comment]
Actually maybe not 100% fixed, looking at my notifications log some were doubled up. But it does stop now at least rather than sending them until I kill the server.

**Comment by mxcl** [comment]
Yeah ok, this seems to possibly the issue, I put a print counter into the code when HTTP2 streams are created and when I use my test case I get up to 66 open connections to APNs.

A google suggests 15 is the undocumented limit per IP that Apple allow before they start closing the connections.

Also my reading suggests that we are preemptively creating new streams, Apple say you should be able to do 9000 payload writes per second to one connection, I'm just trying to write 66 here within a few seconds.

Anyway, if there's any feedback I can take it into account as I attempt to fix this. Thanks.
